pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -0,0 +1,217 @@
+		IsController: true,
+		OwnerType:    &cloudingressv1alpha1.PublishingStrategy{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcilePublishingStrategy implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcilePublishingStrategy{}
+
+// ReconcilePublishingStrategy reconciles a PublishingStrategy object
+type ReconcilePublishingStrategy struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a PublishingStrategy object and makes changes based on the state read
+// and what is in the PublishingStrategy.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling PublishingStrategy")
+
+	// Fetch the PublishingStrategy instance
+	instance := &cloudingressv1alpha1.PublishingStrategy{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile will handle
+	// 1. Set the cluster API to Internal
+	// 2. Set the cluster API to External (Internet-facing)
+	// 3. Set the default ingress (application) to Internal
+	// 4. Set the default ingress (application) to External (Internet-facing)
+
+	// get region
+	region, err := utils.GetClusterRegion(r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	// Secret should exist in the same namespace Account CR's are created
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: config.AWSSecretName,
+		NameSpace: 	config.OperatorNamespace,
+		AwsRegion: 	region
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AWS client")
+		return reconcile.Result{}, err
+	}
+
+	// To make API server non-public
+	// 1. Delete ext NLB for port 6443/TCP
+	// 2. Update api.<cluster-domain> DNS record to point to internal NLB
+	if instance.Spec.DefaultAPIServerIngress.Listening == cloudingressv1alpha1.Internal {
```

Comment:
I'd like to see a comment about what the goal of this comparison is for. In the line of "If <x>, then we have to <y>" where x is a statement of customer request and y is the high level action we're taking. For example:

> // If the CR is wanting the default API server to be internal facing only, we need to: <list of steps in sequence to accomplish the goal>

The reason is that it helps folks to get inside your head and set some expectations for the block of code to come. People will mentally be checking things off the <list of steps> as they read through the code.

Reply:
This is a great point.